### DISCUSSION TITLE
Installing current master branch of the RDKit from GitHub before next release

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#6595] [Q&A]
Is there a way to install the master branch of the RDKit from GitHub? In a Jupyter Notebook, I'm documenting a [feature to be released in the future](https://github.com/rdkit/rdkit/pull/6080), so I'd like to get the current version of the master branch to run that new code in the notebook. In case it's relevant, I just need new Python functionality, not any new C++ functionality.

P.S. I tried a venv (not conda) approach to [installing from GitHub](https://stackoverflow.com/questions/15268953/how-to-install-python-package-from-github#15268990)

```
pip install git+https://github.com/rdkit/rdkit.git#egg=rdkit
```

and got

```
ERROR: rdkit from git+https://github.com/rdkit/rdkit.git#egg=rdkit does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

**Comment by DavidACosgrove** [comment]
Hi @bertiewooster,
I assume that you are still using a macOS development environment.  If so, then I think that the instructions at https://www.rdkit.org/docs/Install.html for Mac installation should work.  The key extra bits are to do 'make install' and then set DYLD_FALLBACK_LIBRARY_PATH.  If you run Jupyter notebook inside your development conda/mamba environment, it should pick up the newly compiled version.  Doing
```
from rdkit import rdBase
print(rdBase.rdkitVersion)
```
should give you 2023.09.pre or something like that to show that you have succeeded.
Dave

**Reply by bertiewooster** [reply]
Hi @DavidACosgrove, thanks. I'll try running the code in my REKit development environment.

**Comment by bp-kelley** [comment]
You can always look at the generic build recipe we use the continuous
integration, it's located here:

https://github.com/rdkit/rdkit/blob/master/.azure-pipelines/mac_build.yml

If might help if you find yourself in a spot of trouble.

Additionally you can use the conda recipe here which creates an installable
conda package.

https://github.com/conda-forge/rdkit-feedstock

Finally, the pypi build system is here:

https://github.com/kuelumbus/rdkit-pypi

But I haven't tried this locally on OSX.

Good luck!
